### PR TITLE
Handle multiple conditions in policies

### LINKED ISSUE #7078: mc policy parser truncate line when s3:delimiter option is used
## Expected Behavior
When you add a policy via `mc policy add X Y file.json`, and then print it with `mc policy list X Y`, the output should be equal to the original policy file. 

## Current Behavior
When uploading a policy containing s3:delimiter option the line is truncated.
Example json file:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "AllowGroupToSeeBucketListInTheConsole",
      "Action": [
        "s3:ListAllMyBuckets",
        "s3:GetBucketLocation"
      ],
      "Effect": "Allow",
      "Resource": [
        "arn:aws:s3:::*"
      ]
    },
    {
      "Sid": "AllowRootAndHomeListingOfCompanyBucket",
      "Action": [
        "s3:ListBucket"
      ],
      "Effect": "Allow",
      "Resource": [
        "arn:aws:s3:::my-company"
      ],
      "Condition": {
        "StringEquals": {
          "s3:prefix": [
            "",
            "home/"
          ],
          "s3:delimiter": [
            "/"
          ]
        }
      }
    }
  ]
}
```

yields:

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "AllowGroupToSeeBucketListInTheConsole",
      "Effect": "Allow",
      "Action": [
        "s3:GetBucketLocation",
        "s3:ListAllMyBuckets"
      ],
      "Resource": [
        "arn:aws:s3:::*"
      ]
    },
    {
      "Sid": "AllowRootAndHomeListingOfCompanyBucket",
      "Effect": "Allow",
      "Action": [
        "s3:ListBucket"
      ],
      "Resource": [
        "arn:aws:s3:::my-company"
      ],
      "Condition": {
        "StringEquals": {
          "s3:delimiter": [
            "/"
          ]
        }
      }
    }
  ]
}
```

Notice how the condition of statement 2 has been truncated.
Note: If you remove the first statement is also truncated but this time it looks like: `"Condition":{"StringEquals":{"s3:prefix":["","home/"]}}`...

## Your Environment
VERSION
2018-12-27T18:33:08Z
MEMORY
Used: 13 MB | Allocated: 82 GB | Used-Heap: 13 MB | Allocated-Heap: 154 MB
PLATFORM
Host: ea250a18f3c9 | OS: linux | Arch: amd64
RUNTIME
Version: go1.10.7 | CPUs: 4

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Handle multiple conditions in policies

## Motivation and Context
We did not handle multiple conditions in policies till now, handle it appropriately.

Fixes #7078

## Regression
No

## How Has This Been Tested?
Added unit tests and also using the policies provided in issue #7078 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by harshavardhana** [comment]
rebased with master PTAL @donatello @balamurugana

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by donatello — DISMISSED** [review]

File: pkg/policy/condition/func_test.go [hunk 1]
```diff
@@ -196,7 +196,14 @@ func TestFunctionsUnmarshalJSON(t *testing.T) {
         "s3:x-amz-metadata-directive": "REPL*"
     },
     "StringEquals": {
-        "s3:x-amz-copy-source": "mybucket/myobject"
+        "s3:x-amz-copy-source": "mybucket/myobject",
```
Comment (balamurugana) [inline pkg/policy/condition/func_test.go:281]: create one more test case than modifying existing one which serves one purpose.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: pkg/policy/condition/func_test.go (hunk 1 above)
Comment (harshavardhana) [reply pkg/policy/condition/func_test.go:281]: done

**Comment by harshavardhana** [comment]
updated with a new test case PTAL @balamurugana @donatello

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by balamurugana — APPROVED** [review]

**Comment by harshavardhana** [comment]
@donatello can you reapprove ?

**Review by donatello — APPROVED** [review]

**Pull request merged**